data/src/test/java/org/apache/iceberg/data/TestReadProjection.java

```diff
@@ -76,20 +78,21 @@ public void testSpecialCharacterProjection() throws Exception {
 
     Record full = writeAndRead("special_chars", schema, schema, record);
 
-    Assert.assertEquals(
-        "Should contain the correct id value", 34L, (long) full.getField("user id"));
-    Assert.assertEquals(
-        "Should contain the correct data value",
-        0,
-        Comparators.charSequences().compare("test", (CharSequence) full.getField("data%0")));
+    Assertions.assertThat((long) full.getField("user id"))
+        .isEqualTo(34L)
+        .describedAs("Should contain the correct id value");
```

Comment:
this should be rather `.as(..)` and should come before the final assertion. Also please update all the other places in this PR